src/template.yml

```diff
@@ -830,12 +825,15 @@ Resources:
                 - s3:PutEncryptionConfiguration
                 - s3:PutBucketPolicy
                 - s3:PutBucketPublicAccessBlock
-              Resource: "arn:aws:s3:::adf-shared-modules-*"
+              # This must match BucketNamePrefix of the SharedModulesBucket resource
+              Resource: !Sub "arn:${AWS::Partition}:s3:::adf-shared-modules-${DeploymentAccountMainRegion}-*"
```

Comment:
When updating, I noticed that the DeploymentAccountMainRegion is not enforced.
If you forget to specify it, it will create a new bucket without the region in its name.
So far, that has not caused any issues. However, when we introduce this change, it will.

So either we should enforce the DeploymentAccountMainRegion in the stack, and make it clear that users should not forget to specify the region. Or we should implement a way to lookup the previous value.
Unfortunately the SAR does not show the values used previously when deploying the same template.